server/src/controllers/group-service.ts

```diff
@@ -91,4 +92,42 @@ module.exports = class GroupService {
       });
     });
   }
+
+  deleteGroup(groupID: number) {
+    return new Promise((resolve, reject) => {
+      // get list of group expenses by group id
+      GroupExpense.getGroupExpenseByGroupId(groupID, (err: any, res: any) => {
```

Comment:
super janky delete group cascading endpoint, did not extensively test due to having to add all related components and relationships, but seemed to delete everything on the trial I did. May need to update later